### PR TITLE
[DO NOT MERGE]: posix shutdown investigation/hacks

### PR DESCRIPTION
**Pull request by dagar**
- review posix shutdown
 - review px4 client/server locking (and cleanup)

What's the relation to unit tests failing with the new px4 work queue? https://github.com/PX4/Firmware/pull/11261

### CODE REVIEW

**Comment by julianoes** [comment]
I don't understand the `dyn` test failure:
```
      Start 31: dyn
31/34 Test #31: dyn ..............................***Failed  Required regular expression not found.Regex=[1: PASSED
]  0.02 sec
ERROR [dyn] ./examples__dyn_hello.px4mod: cannot open shared object file: No such file or directory
INFO  [px4] Creating symlink /home/julianoes/src/Firmware/ROMFS/px4fmu_test -> /home/julianoes/src/Firmware/build/px4_sitl_test/tmp/etc

______  __   __    ___ 
| ___ \ \ \ / /   /   |
| |_/ /  \ V /   / /| |
|  __/   /   \  / /_| |
| |     / /^\ \ \___  |
\_|     \/   \/     |_/

px4 starting.

57 WARNING: setRealtimeSched failed (not run as root?)
INFO  [px4] Calling startup script: /bin/sh /home/julianoes/src/Firmware/posix-configs/SITL/init/test/test_dyn_hello 0
Shutting down
```

**Comment by dagar** [comment]
> I don't understand the `dyn` test failure:
> 
> ```
>       Start 31: dyn
> 31/34 Test #31: dyn ..............................***Failed  Required regular expression not found.Regex=[1: PASSED
> ]  0.02 sec
> ERROR [dyn] ./examples__dyn_hello.px4mod: cannot open shared object file: No such file or directory
> INFO  [px4] Creating symlink /home/julianoes/src/Firmware/ROMFS/px4fmu_test -> /home/julianoes/src/Firmware/build/px4_sitl_test/tmp/etc
> 
> ______  __   __    ___ 
> | ___ \ \ \ / /   /   |
> | |_/ /  \ V /   / /| |
> |  __/   /   \  / /_| |
> | |     / /^\ \ \___  |
> \_|     \/   \/     |_/
> 
> px4 starting.
> 
> 57 WARNING: setRealtimeSched failed (not run as root?)
> INFO  [px4] Calling startup script: /bin/sh /home/julianoes/src/Firmware/posix-configs/SITL/init/test/test_dyn_hello 0
> Shutting down
> ```

The path seems to be wrong now.

File: platforms/posix/src/main.cpp [hunk 1]
```diff
@@ -436,8 +431,8 @@ void sig_fpe_handler(int sig_num)
 	printf("\nfloating point exception\n");
 	PX4_BACKTRACE();
 	fflush(stdout);
-	px4_daemon::Pxh::stop();
-	_exit_requested = true;
+
+	px4_systemreset(false);
```
Comment (mcsauder) [inline platforms/posix/src/main.cpp]: @dagar , did you want this to be `px4_system_reset`?

**Comment by MaEtUgR** [comment]
I did a quick test because I though it might fix #11027 but doesn't but it looks like a good start.

**Comment by dagar** [comment]
> I did a quick test because I though it might fix #11027 but doesn't but it looks like a start.

It's intended as a start. There's a lot more to do.

**Pull request closed without merging**